agent/api/task/task.go

```diff
@@ -836,15 +887,55 @@ func (task *Task) collectLogOptions() (map[string]map[string]string, error) {
 		hostConfig := &dockercontainer.HostConfig{}
 		err := json.Unmarshal([]byte(*container.DockerConfig.HostConfig), hostConfig)
 		if err != nil {
-			return nil, errors.Wrapf(err, "unable to decode host config of container %s", container.Name)
+			return errors.Wrapf(err, "unable to decode host config of container %s", container.Name)
 		}
 
 		if hostConfig.LogConfig.Type == firelensDriverName {
-			containerToLogOptions[container.Name] = hostConfig.LogConfig.Config
+			if containerToLogOptions[container.Name] == nil {
+				containerToLogOptions[container.Name] = make(map[string]string)
+			}
+			for k, v := range hostConfig.LogConfig.Config {
+				containerToLogOptions[container.Name][k] = v
+			}
 		}
 	}
 
-	return containerToLogOptions, nil
+	return nil
+}
+
+// collectFirelensLogEnvOptions collects all the log secret options. Each secret log option will have a value
+// of a config file variable (e.g. "${config_var_name}") and we will pass the secret value as env to the firelens
+// container and it will resolve the config file variable from the env.
+// Each config variable name has a format of log-option-key_container-name. We need the container name because options
+// from different containers using awsfirelens log driver in a task will be presented in the same firelens config file.
+func (task *Task) collectFirelensLogEnvOptions(containerToLogOptions map[string]map[string]string, firelensConfigType string) error {
```

Comment:
we have the function above `initializeFirelensResource()` which initializes the log options map and then populates it with`task.collectFirelensLogOptions(containerToLogOptions)`.

Is this doing some of that same work? 

I'd like to see these two functions more clearly differentiated, with any common functionality pulled out.  This will help me as a reader understand how initialize and collect work together.

Reply:
not sure what's your concern here.. initializeFirelensResource is the caller of collectFirelensLogOptions. i will remove the map initialization in the callee since that seems to make you confused..

Reply:
updated